Clinical-Genomics/cgbeacon2
# BND variants and other fixes

### fix #34

List of things included in this PR (+ tests)

### Fixed
- Accept variants annotated with build GRCh37 and GRCh38 (`chrN`) instead of just `N` (as in hg19)
- Improved calculation of structural variants end coordinates

### Changed
- Renamed SNV and SV demo VCF files
- Coordinate Range queries allowing fuzzy positions

### Added
- Demo VCF file containing BND SV variants
- Save BND variants to database (introduced additional mateName key/values)
- Query BND variants using mateName

## How to test it:
1. Populate a demo database, that now will include also BND variants:
`cgbeacon2 add demo`

1. Run the server:
`cgbeacon2 run`

1. Launch the following query;
```
curl -X POST \
    localhost:5000/apiv1.0/query \
    -H 'Content-Type: application/json' \
    -H 'Accept: application/json' \
    -d '{"referenceName": "17",
    "mateName": "2",
    "variantType" : "BND",
    "startMin": 198000,
    "startMax": 200000,
    "referenceBases": "A",
    "assemblyId": "GRCh37",
    "includeDatasetResponses": "HIT"}'
```
- The server should return exists: True


### Review:
